AlphaWallet/UI/TokenObject+UI.swift

```diff
@@ -57,24 +57,35 @@ class RPCServerImageFetcher {
     }
 }
 
-extension TokenObject {
-    fileprivate static let numberOfCharactersOfSymbolToShowInIcon = 4
+extension TokenImageble {
 
     func icon(withSize size: GoogleContentSize) -> Subscribable<TokenImage> {
-        return TokenImageFetcher.instance.image(forToken: self, size: size)
+        let name = symbol.nilIfEmpty ?? name
+        return TokenImageFetcher.instance.image(contractAddress: contractAddress, server: server, name: name, type: type, balance: firstNftAsset, size: size)
     }
 }
 
-extension Token {
-
-    func icon(withSize size: GoogleContentSize) -> Subscribable<TokenImage> {
-        let name = symbol.nilIfEmpty ?? name
-        let nftBalance = nftBalanceValue.first
+protocol TokenImageble {
```

Comment:
I was going to suggest `HasTokenImage`, but the protocol looks like it isn't about that. Maybe add a TODO to rename it?

Reply:
`HasTokenImage ` That is mutch better